rskj-core/src/main/java/co/rsk/peg/federation/FederationStorageProviderImpl.java

```diff
@@ -208,12 +208,47 @@ public void setPendingFederation(PendingFederation federation) {
         pendingFederation = federation;
     }
 
+    @Override
+    public void clearPendingFederation() {
+        setPendingFederation(null);
+    }
+
+    @Override
+    public Optional<Federation> getProposedFederation(FederationConstants federationConstants, ActivationConfig.ForBlock activations) {
+        if (!activations.isActive(RSKIP419)) {
+            return Optional.empty();
+        }
+
+        if (proposedFederation != null || isProposedFederationSet) {
+            return Optional.ofNullable(proposedFederation);
+        }
+
+        proposedFederation = bridgeStorageAccessor.getFromRepository(
+            PROPOSED_FEDERATION.getKey(),
+            data -> {
+                if (data == null) {
+                    return null;
+                }
+                // storage version should be always present for non-null proposed federation
+                Optional<Integer> storageVersion = getStorageVersion(PROPOSED_FEDERATION_FORMAT_VERSION.getKey());
```

Comment:
Should we check that the StorageVersion is present?

Reply:
after rskip186 storage version is always saved too. So if the saved version is null, is because the saved fed is null. And that's covered when checking for null data